test/test_helper.rb

```diff
@@ -214,19 +205,41 @@ def obj.run(*)
       obj
     end
 
+    def logs_message_if_captured
+      unless log_to_real_fds?
+        " Logs:\n#{@logger_stream.string}"
+      end
+    end
+
+    def mock_output_stream
+      if log_to_real_fds?
+        $stdout
+      else
+        @mock_output_stream
+      end
+    end
+
     private
 
-    def log_to_stderr?
+    def log_to_real_fds?
       ENV["PRINT_LOGS"] == "1"
     end
 
     def logging_assertion
-      if log_to_stderr?
+      if log_to_real_fds?
         $stderr.puts("\033[0;33mWARNING: Skipping logging assertions while logs are redirected to stderr\033[0m")
       else
         @logger_stream.rewind
         yield @logger_stream.read
       end
     end
+
+    def stdout_assertion
+      if log_to_real_fds?
+        $stderr.puts("\033[0;33mWARNING: Skipping stream assertions while logs are redirected to stderr\033[0m")
+      else
+        yield @mock_output_stream.string
```

Comment:
This does seem simpler than the rewind/read we do in the original method above. If you're already going to make another commit, would you mind making this improvement there too?